pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -583,6 +583,9 @@ protected synchronized boolean trySendMessagesToConsumers(ReadType readType, Lis
         if (needTrimAckedMessages()) {
             cursor.trimDeletedEntries(entries);
         }
+        if (readType == ReadType.Replay) {
+            entries.removeIf(entry -> !redeliveryMessages.contains(entry.getLedgerId(), entry.getEntryId()));
+        }
```

Comment:
I have tried to remove these lines to check the newly added test `concurrentlyReceiveDeliveriedMessages`

It always gets passed with `invocationCount = 100`

Reply:
I have modified the test to make that the probability of problem occuring is greater than 50%.